sklearn/multiclass.py

```diff
@@ -161,8 +161,8 @@ def predict_proba(self, X):
             ensure_2d=False,
             reset=False,
         )
-
-        return np.repeat([np.hstack([1 - self.y_, self.y_])], _num_samples(X), axis=0)
+        y_ = self.y_.astype(np.float64)
```

Comment:
`np.float64` or `X.dtype`?

Reply:
In general, I think `predict_proba` outputs `np.float64` and does not depend on `X.dtype`:

```python
import numpy as np
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression

X, y = make_classification(random_state=0)
X = X.astype(np.float32)
log_reg = LogisticRegression().fit(X, y)

log_reg.predict_proba(X).dtype
# dtype('float64')
```

For the specific `OneVsRestClassifier` case, `X.dtype` can be integer and we still want `predict_proba` to output floats.